examples/computer_vision/vision_transformer/det_callback.py

```diff
@@ -0,0 +1,115 @@
+            else:
+                # can that even happen?
+                pass
+
+    def process_log(self, log):
+        metric_type = self._metric_type(log)
+        metrics = log
+
+        if self.filter_metrics is not None:
+            metrics = {}
+            for k, v in log.items():
+                if any(m in k for m in self.filter_metrics) is True:
+                    metrics[k] = v
+
+        return metric_type, metrics
+
+    def _metric_type(self, d):
+        for k, v in d.items():
+            if k.startswith("eval"):
+                return "eval"
+            elif k.startswith("test"):
+                return "test"
+            else:
+                return "train"
+
+    def on_save(self, args: TrainingArguments, state: TrainerState, control: TrainerControl, **kwargs):
+        info = det.get_cluster_info()
+        assert info is not None
+        if state.is_world_process_zero:
+            save_path = os.path.join(args.output_dir, f"checkpoint-{state.global_step}")
+            det_checkpoint__metadata = {"steps_completed": state.global_step, "trial_id": info.trial.trial_id}
+
+            if self.tokenizer_options is not None:
+                det_checkpoint__metadata['tokenizer_options'] = self.tokenizer_options
+
+            if self.tokenizer is not None:
+                self.tokenizer.save_pretrained(os.path.join(save_path, "tokenizer"))
+
+            if self.user_checkpoint_metadata is not None:
+                self._on_save(save_path)
+
+            self.core_context.checkpoint.upload(save_path, det_checkpoint__metadata)
+
+    def _on_save(self, save_path):
+        '''
+        User-defined saving of objects from self.checkpoint_metadata under save_path.
+        After objects are saved, det handles uploading and downloading objects to/from selected storage.
+        '''
+        pass
+
+    def load_last_checkpoint(self, args):
+        info = det.get_cluster_info()
+        assert info is not None
+        latest_checkpoint = info.latest_checkpoint
+
+        if latest_checkpoint is not None:
+            metadata = self.core_context.checkpoint.get_metadata(latest_checkpoint)
+            prev_trial_id = metadata["trial_id"]
+            trial_id = info.trial.trial_id
+            if trial_id != prev_trial_id:
+                resume_step = 0
+            else:
+                resume_step = metadata['steps_completed']
+            checkpoint_path = os.path.join(args.output_dir, f"checkpoint-{resume_step}")
+            self.core_context.checkpoint.download(latest_checkpoint, checkpoint_path)
+
+    def on_epoch_end(self, args: TrainingArguments, state: TrainerState, control: TrainerControl, **kwargs):
+        if self.core_context.preempt.should_preempt():
+            return
+
+
+def override_training_args(training_args):
+    hparams = det.get_cluster_info().trial.hparams
+    for k, v in hparams.items():
+        if hasattr(training_args, k):
+            training_args.k = v
+    return training_args
```

Comment:
To fully support Trainer API integration we need to add Hp scheduler (e.g., [Ray](https://github.com/huggingface/transformers/blob/main/src/transformers/integrations.py#L171))